tests/ssr.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect } from 'react'
+import { act, screen } from '@testing-library/react'
+import { renderToString } from 'react-dom/server'
+import { describe, expect, it } from 'vitest'
+import { create } from 'zustand'
+
+const describeTest = React.version.startsWith('18') ? describe : describe.skip
+
+describeTest('ssr behavior with react 18', () => {
```

Comment:
Let's use `describe.skipIf`

```suggestion
describe.skipIf(!React.version.startsWith('18'))('ssr behavior with react 18', () => {
```